pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -66,14 +60,25 @@ func NewHealthChecker(monCluster *Cluster) *HealthChecker {
 			MonOutTimeout = monTimeout
 		}
 	}
+}
 
+func updateMonInterval(monCluster *Cluster, h *HealthChecker) {
 	checkInterval := monCluster.spec.HealthCheck.DaemonHealth.Monitor.Interval
 	// allow overriding the check interval
 	if checkInterval != nil {
 		logger.Infof("ceph mon status in namespace %q check interval %q", monCluster.Namespace, checkInterval.Duration.String())
 		h.interval = checkInterval.Duration
 	}
+}
 
+// NewHealthChecker creates a new HealthChecker object
+func NewHealthChecker(monCluster *Cluster) *HealthChecker {
+	h := &HealthChecker{
+		monCluster: monCluster,
+		interval:   HealthCheckInterval,
+	}
+	updateMonTimeout(monCluster)
```

Comment:
Instead of updating the timeouts here, how about before line 88 below before the `select` statement? Then it will be re-evaluated before each health check instead of doing it inside the `checkHealth()` method